src/viur/core/modules/file.py

```diff
@@ -869,6 +938,109 @@ def download(self, blobKey: str, fileName: str = "", download: bool = False, sig
 
         raise errors.Redirect(signedUrl)
 
+    @exposed
+    def serve(
+        self,
+        key: str,
+        size: t.Optional[int] = None,
+        filename: t.Optional[str] = None,  # random string with .ext
+        options: str = "",
+        download: bool = False,
+    ):
```

Comment:
Refactored this. The type annotations are used by ViUR 3 to parse specific types. @akelch the original version looked a little bit old, I hope that I've refactored it correctly.